### PR TITLE
[release] Handle exceptions on requests failure when retrieving wheel

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: rickyyx <rickyx@anyscale.com>

## Why are these changes needed?

Exceptions currently not caught when getting wheel URL, causing tests to fail

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [X] This PR is not tested :(

### CODE REVIEW

**Comment by rickyyx** [comment]
Any thoughts on testing? I could trigger a release tests to make sure it doesn't break anything

File: release/ray_release/util.py [hunk 1]
```diff
@@ -42,7 +42,11 @@ def dict_hash(dt: Dict[Any, Any]) -> str:
 
 
 def url_exists(url: str) -> bool:
-    return requests.head(url, allow_redirects=True).status_code == 200
+    try:
+        return requests.head(url, allow_redirects=True).status_code == 200
+    except requests.exceptions.RequestException as e:
+        logger.warning(f"Failed to get requests: {e}")
```
Comment (simon-mo) [inline release/ray_release/util.py:48]:
```suggestion
        logger.exception(f"Failed to check url exists: {url}")
```

**Review by simon-mo — APPROVED** [review]

**Comment by simon-mo** [comment]
> Any thoughts on testing?
Good thinking, alternatively you can try trigger a release test pointing to a non-existent host like `wheel_url = "localhost:1234..."` which will for sure throw

File: release/ray_release/util.py [hunk 2]
```diff
@@ -42,7 +42,11 @@ def dict_hash(dt: Dict[Any, Any]) -> str:
 
 
 def url_exists(url: str) -> bool:
-    return requests.head(url, allow_redirects=True).status_code == 200
+    try:
+        return requests.head(url, allow_redirects=True).status_code == 200
+    except requests.exceptions.RequestException:
+        logger.exception(f"Failed to check url exists: {url}")
+        return False
```
Comment (krfricke) [inline release/ray_release/util.py:49]: Can we log the exception here, too?

```suggestion
    except requests.exceptions.RequestException as e:
        logger.exception(f"Failed to check if URL exists ({url}): {e}")
        return False
```

**Review by krfricke — COMMENTED** [review]
For testing, maybe just add a unit test and try to resolve an invalid URL?

File: release/ray_release/util.py (hunk 2 above)
Comment (krfricke) [reply release/ray_release/util.py:49]: `logger.exception` should be enough

**Review by krfricke — APPROVED** [review]
I've added a unit test and will merge after CI pass

**Pull request merged**